FrontEnd/src/Notifications/NotificationsPage.jsx

```diff
@@ -12,16 +12,13 @@ function NotificationsPage({ userData, vendorID }) {
       return;
     }
     const unsubscribe = fetchNotifications(userID, (notificationData) => {
-      const userNotifications = notificationData.filter(
-        notification => notification.userID === userID
-      );
+      const userNotifications = notificationData
```

Comment:
Why not pass `notificationData` directly to `setUserNotificationData`, instead of defining a new variable?